go/vt/sqlparser/token.go

```diff
@@ -1027,11 +1031,98 @@ func (tkn *Tokenizer) scanMySQLSpecificComment() (int, []byte) {
 		}
 		tkn.consumeNext(buffer)
 	}
-	_, sql := ExtractMysqlComment(buffer.String())
-	tkn.specialComment = NewStringTokenizer(sql)
+	version, sql := ExtractMysqlComment(buffer.String())
+	commentVersion, err := convertCommentVersion(version)
+	if err != nil {
+		tkn.Error(err.Error())
+		return tkn.Scan()
+	}
+
+	isCommentVersionHigher := false
+
+	if len(commentVersion) == 3 {
+		mysqlVersion, err := convertMySQLVersion(MySQLVersion)
+		if err != nil {
+			tkn.Error(err.Error())
+			return tkn.Scan()
+		}
+		for i := 0; i < 3; i++ {
+			if mysqlVersion[i] > commentVersion[i] {
+				break
+			} else if mysqlVersion[i] < commentVersion[i] {
+				isCommentVersionHigher = true
+				break
+			}
+		}
+	}
+
+	// Only add the special comment to the tokenizer if the version of MySQL is higher or equal to the comment version
+	if !isCommentVersionHigher {
+		tkn.specialComment = NewStringTokenizer(sql)
+	}
 	return tkn.Scan()
 }
 
+func convertMySQLVersion(version string) ([]int, error) {
+	var res = make([]int, 3)
+	idx := 0
+	val := ""
+	for _, c := range version {
+		if c <= '9' && c >= '0' {
+			val += string(c)
+		} else if c == '.' {
+			v, err := strconv.Atoi(val)
+			if err != nil {
+				return nil, err
+			}
+			val = ""
+			res[idx] = v
+			idx++
+			if idx == 3 {
+				break
+			}
+		} else {
+			break
+		}
+	}
+	if val != "" {
+		v, err := strconv.Atoi(val)
+		if err != nil {
+			return nil, err
```

Comment:
converting mysql_version to comment version will be easier to convert and compare. This complex logic can be avoided.

Reply:
I'm not seeing how converting mysql_version to comment version is much easier. Could you expand on your thinking?